Cluster Selection Modal

timobraz opened the pull request:
Created cluster-selection and cluster-autocomplete components that open a modal pictured below for selecting clusters from the available cluster list.
![image](https://github.com/user-attachments/assets/7871a140-077c-48c7-a24e-3d59e567b2dd)
![image](https://github.com/user-attachments/assets/55d7af23-8396-4e46-b6a6-40886d513764)

timobraz committed:
feat: Added cluster selection panel upon clicking cluster input window

timobraz committed:
fix: warning field not showing up

timobraz committed:
test: added cluster auto-complete test

kunwp1 reviewed (changes requested):
Overall, great work! The implementation looks promising, but I have a few suggestions to improve it further:

1. The drop-down menu currently shows the cluster ID and cluster name in two separate columns. However, the alignment does not look quite right. To improve readability, please right-align the cluster IDs in column 1 and left-align the cluster names in column 2.
<img width="538" alt="Screenshot 2024-09-05 at 11 17 13 PM" src="https://github.com/user-attachments/assets/2454209e-c147-4032-82ef-7acdcd3b9f49">


2. The "Cluster Information" section shows the "Created At" timestamp, but it would be more informative to display the time zone as well. For reference, check the "Admins" > "Executions" tab, where the time zone is explicitly shown. Consider either displaying the time zone directly in the component or adding a tooltip that appears when the user hovers over the "Created At" section.


3. The current behavior of the "Cancel" button is unintuitive. When a user selects a new cluster ID (e.g., cluster ID 2) from the drop-down menu and then clicks "Cancel," the selected cluster ID is still being applied to the operator property panel. Instead, the previously selected value (e.g., cluster ID 1) should remain unchanged if the user cancels the selection.

timobraz committed:
fix:alignment, cancel modal, timezones

timobraz commented:
Fixed alignment and logic changes. Added tooltip that displays full date with time-zone. Fixed cancel modal behavior

![image](https://github.com/user-attachments/assets/ae2b2f05-9cfc-4a3b-841a-5d4d7cc450de)

kunwp1 reviewed (approved):
LGTM!
